### DISCUSSION TITLE
REST api : Unable to define filters to data from api

### DISCUSSION THREAD

**Discussion by nathaponb** [Budibase/budibase#5467] [Help] [answered]
Apparently by using Data Provider to retrieve an API data is lack of filtering functionality compare to SQL/Budibase Databases approach,
Data is fetched and could be display accordingly but unable to filtering by its corresponding column names.

![budibase](https://user-images.githubusercontent.com/75414640/163791703-c95f5ecd-1980-431f-af61-a6a6b2464a08.png)

**Comment by bjornclauw** [comment]
For me this is working as expected. Is your API returning a schema?

![image](https://user-images.githubusercontent.com/32811735/163810218-54413577-8185-4fe8-ae83-5050554b093b.png)

Sometimes when i update the transformer i need te resend the query and save a couple of times for the schema to 'stick'.

(layout of my testapp)
![image](https://user-images.githubusercontent.com/32811735/163812136-5748ad86-e35e-470d-9d1b-e14e65f04ad3.png)

![image](https://user-images.githubusercontent.com/32811735/163814441-642e46de-1a42-454e-be1f-8a61be12c283.png)

**Reply by nathaponb** [reply]
yes it returns schema of an array of objects like so

![Untitled2](https://user-images.githubusercontent.com/75414640/163914276-4a8a7670-7338-4b5e-892e-1a5db3802c48.png)

![Untitled1](https://user-images.githubusercontent.com/75414640/163914040-1c7fac0f-ce85-4890-b5a9-7ea01e616397.png)

![Untitled3](https://user-images.githubusercontent.com/75414640/163914500-f38d0847-9b00-4ab9-96a8-fbc927cd4ce4.png)

**Comment by bjornclauw** [comment]
![Screenshot_20220419-053831_Samsung Internet](https://user-images.githubusercontent.com/32811735/163915484-b9698f2d-3b10-45e9-b13a-7774f4e14911.jpg)

The schema auto populates after it receives the fields from the transformer. So instead of `return data` you need something like the code above.
(Peronally, i use `return data` and push the read button. Then i take the raw data from the field below and copypaste it in an online json editor to lookup the fields I need -> data.object1[0].value1)

**Reply by nathaponb** [reply]
It's a little bit tricky for me, i couldn't figure out why would i need to use Transformer, because API response "Array of objects" is expected and recommended data structure from Budibase to make a Schema out of it. 

API response data structure
```
[
  {id: 0, description: "hello"},
  {id: 1, description: "hello again"},
  ...
]
```

**Comment by Kuuchuu** [comment, marked as answer]
I experienced this issue due to a bug in Budibase somewhere between v1.0.73 and v1.0.90(?). What version are you currently running? Is the app itself up to date/the same version as your Budibase install?

**Comment by nathaponb** [comment]
@Kuuchuu
v.1.0.73 seem to be the latest version available?

![Untitled4](https://user-images.githubusercontent.com/75414640/163940111-b2689923-52d4-4a97-9003-0731e5055425.png)

**Comment by nathaponb** [comment]
@Kuuchuu 
My Budibase Installation is up to date. but when i create an app and check its version it turns out to be v.1.0.73

![Untitled8](https://user-images.githubusercontent.com/75414640/163953442-91641a11-b7a6-464f-b7ec-4030b31287f5.png)

**Reply by melohagan** [reply]
@hellonathapon Are you self hosting? You can use the budi CLI to make sure you're using the latest docker-compose file https://github.com/Budibase/budibase/discussions/5169

**Comment by nathaponb** [comment]
It seems like older Budibase versions lack of filtering functionality, so I
fixed it by clean installing the latest version via Docker compose as a
reply suggested to me and it works.
I notice you mention me but couldn't find it in #5467
<https://github.com/Budibase/budibase/discussions/5467?sort=top> comment so
i send this email, hope it helps.
